pkg/mcp/tools.go

```diff
@@ -0,0 +1,110 @@
+type ToolsetType string
+
+const (
+	// ToolsetCore represents the core toolset with basic operations
+	ToolsetCore ToolsetType = "core"
+)
+
+type Toolsets struct {
+	CoreToolset CoreToolsetHandler
+}
+
+type CoreToolsetHandler interface {
+	GetOrganization(name string) (string, error)
+	ListProjects(orgName string) (string, error)
+}
+
+// Helper functions to create JSON Schema definitions
+func stringProperty(description string) map[string]any {
+	return map[string]any{
+		"type":        "string",
+		"description": description,
+	}
+}
+
+func enumProperty(description string, values []string) map[string]any {
+	return map[string]any{
+		"type":        "string",
+		"description": description,
+		"enum":        values,
+	}
+}
+
+func createSchema(properties map[string]any, required []string) map[string]any {
+	schema := map[string]any{
+		"type":       "object",
+		"properties": properties,
+	}
+	if len(required) > 0 {
+		schema["required"] = required
+	}
+	return schema
+}
+
+func (t *Toolsets) Register(s *mcp.Server) {
+	if t.CoreToolset != nil {
+		mcp.AddTool(s, &mcp.Tool{
+			Name:        "get_organization",
+			Description: "Get information about organizations. If no name is provided, lists all organizations.",
+			InputSchema: createSchema(map[string]any{
+				"name":          stringProperty("Optional: specific organization name to retrieve"),
+				"output_format": enumProperty("Output format for the results", []string{"json", "table"}),
+			}, []string{}),
+		}, func(ctx context.Context, req *mcp.CallToolRequest, args struct {
+			Name         string `json:"name"`
+			OutputFormat string `json:"output_format"`
+		}) (*mcp.CallToolResult, map[string]string, error) {
+			msg, err := t.CoreToolset.GetOrganization(args.Name)
+			if err != nil {
+				return nil, nil, err
+			}
+			contentBytes, err := json.Marshal(msg)
+			if err != nil {
+				return nil, nil, err
+			}
+			stringContent := string(contentBytes)
+			return &mcp.CallToolResult{
+				Content: []mcp.Content{
+					&mcp.TextContent{Text: stringContent},
+				},
+			}, map[string]string{"message": stringContent}, nil
+		})
+		mcp.AddTool(s, &mcp.Tool{
+			Name:        "list_projects",
+			Description: "List all projects in an organization",
+			InputSchema: createSchema(map[string]any{
+				"org_name": stringProperty("Organization name"),
+			}, []string{"org_name"}),
+		}, func(ctx context.Context, req *mcp.CallToolRequest, args struct {
+			OrgName string `json:"org_name"`
+		}) (*mcp.CallToolResult, map[string]string, error) {
+			result, err := t.CoreToolset.ListProjects(args.OrgName)
+			if err != nil {
+				return nil, nil, err
+			}
+			contentBytes, err := json.Marshal(result)
+			if err != nil {
+				return nil, nil, err
+			}
+			stringContent := string(contentBytes)
+			return &mcp.CallToolResult{
+				Content: []mcp.Content{
+					&mcp.TextContent{Text: stringContent},
+				},
+			}, map[string]string{"message": stringContent}, nil
+		})
```

Comment:
Shall we move the tool registrations for dedicated functions? This is to keep the function more readable
E.g.

```suggestion
		RegisterListProjects(mcp)
		RegisterGetProject(mcp)
```